packages/mui-joy/src/Autocomplete/AutocompleteProps.ts

```diff
@@ -284,6 +284,14 @@ type AutocompleteOwnProps<
     variant?: OverridableStringUnion<VariantProp, AutocompletePropsVariantOverrides>;
   };
 
+/**
+ * AutocompleteProps signature:
+ * @template T
+ * @param {string | object} T The option structure, must be a string or a object with { label: string }
```

Comment:
Should it be
```suggestion
 * @param {string | object} T The option structure, must be a string or a object
```
to avoid the misunderstanding that it's not necessary for option objects to have `label` property?

Reply:
@ZeeshanTamboli I put this comment because of the options documentation itself, isn't that right?

![image](https://user-images.githubusercontent.com/52859409/216963201-52fc88ec-b2ce-496a-bca2-08aed67268a8.png)

Reply:
It's by default, but it's not a must.

Reply:
```suggestion
 * @param {string | object} T The option structure, must be a string or an object (by default, only accepts objects with { label: string } )
```

This is true, It's by default and may cause a misunderstanding, do you think it's better to omit this information here or add this note about the default structure?

Reply:
I don't have a strong opinion. I'll let @siriwatknp decide.

Reply:
Thanks @ZeeshanTamboli for the help!